### PR TITLE
Add Lie groups functional library

### PR DESCRIPTION
**Pull request by fantaosha**
## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have completed my CLA (see **CONTRIBUTING**)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: theseus/geometry/functional/lie_group.py [hunk 1]
```diff
@@ -46,7 +46,8 @@ def op(
         ) -> torch.Tensor:
             if jacobians is not None:
                 check_jacobians_list(jacobians)
-                jacobians.append(jop_autograd_fn(input)[0])
+                jacobians_op = jop_autograd_fn(input)[0]
```
Comment (luisenp) [inline theseus/geometry/functional/lie_group.py:49]: Why this change? Seems unnecessary. Maybe this was used during debugging?

**Review by luisenp — APPROVED** [review]
LGTM. Two comments, but the one about the order `op, jop` we can address in a separate PR once all the ops are finished.

File: theseus/geometry/functional/lie_group.py [hunk 2]
```diff
@@ -59,3 +60,43 @@ def op_no_jop(input: torch.Tensor) -> torch.Tensor:
...
+                for jacobian in jacobians_op:
+                    jacobians.append(jacobian)
+            return op_autograd_fn(input0, input1)
+
+        def jop(
+            input0: torch.Tensor, input1: torch.Tensor
+        ) -> Tuple[List[torch.Tensor], torch.Tensor]:
+            return jop_autograd_fn(input0, input1)
+
+        return op, jop
```
[the remark this answers is not in the document]
Comment (fantaosha) [reply theseus/geometry/functional/lie_group.py:96]: `op` and `jop` are functions. In line 94, note that we return `jacobians, errors`

[the remark this answers is not in the document]
Comment (luisenp) [reply theseus/geometry/functional/lie_group.py:96]: Ah sorry, sounds good. Thanks for clarifying.

File: theseus/geometry/functional/lie_group.py (hunk 1 above)
Comment (fantaosha) [reply theseus/geometry/functional/lie_group.py:49]: > Why this change? Seems unnecessary. Maybe this was used during debugging?

There was a bug in previous code since `jop_autograd_fn(input)[0]` is `List[torch.Tensor]`.

**Review by mhmukadam — APPROVED** [review]

**Review by mhmukadam — APPROVED** [review]
LGTM! Nice work @fantaosha on the new Lie group and nice work @luisenp on consolidating the unit test!
